File: src/main/java/pl/mjedynak/idea/plugins/builder/writer/BuilderWriter.java
package pl.mjedynak.idea.plugins.builder.writer;

import com.intellij.openapi.command.CommandProcessor;
import com.intellij.psi.PsiClass;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import org.jetbrains.annotations.VisibleForTesting;
import pl.mjedynak.idea.plugins.builder.psi.PsiHelper;

public class BuilderWriter {

    @VisibleForTesting
    static final String CREATE_BUILDER_STRING = "Create Builder";

    private BuilderWriter() {
        throw new UnsupportedOperationException("Utility class");
    }

    public static void writeBuilder(@NotNull BuilderContext context, @Nullable PsiClass existingBuilder) {
        CommandProcessor commandProcessor = PsiHelper.getCommandProcessor();
        commandProcessor.executeCommand(
                context.project(), new BuilderWriterRunnable(context, existingBuilder), CREATE_BUILDER_STRING, null);
    }
}

File: src/main/kotlin/io/github/kingg22/kobra/builder/writer/BuilderWriter.kt
package io.github.kingg22.kobra.builder.writer

import com.intellij.psi.PsiClass
import io.github.kingg22.kobra.builder.psi.PsiHelper
import org.jetbrains.annotations.VisibleForTesting

public object BuilderWriter {
    @VisibleForTesting
    public const val CREATE_BUILDER_STRING: String = "Create Builder"

    @JvmStatic
    public fun writeBuilder(context: BuilderContext, existingBuilder: PsiClass?) {
        PsiHelper.commandProcessor.executeCommand(
            context.project,
            BuilderWriterRunnable(context, existingBuilder),
            CREATE_BUILDER_STRING,
            null,
        )
    }
}
